### PR TITLE
Fix timeout handling and require Node.js 10

### LINKED ISSUE #47: Hangs after returning status
### Bug

When I check the status for a URL that is behind VPN - which resolves DNS but does not return any response. `is-reachable` returns false as expected but it hangs after returning the status.

### Actual behavior

`is-reachable` hangs after returning status.

### Expected behavior

`is-reachable` should abort after returning status.

### Debug findings

After some debug, I found that timeout is not respected by **checkHttp** function. `pTimeout` exits after the set timeout but `got` in **checkHttp** function continues to make the request. If the request gives no response, the request timeouts after a while and retries multiple times.

### Solution
Currently `is-reachable` is using got@v9.6.0 which had some issues with timeout as described in [got#997](https://github.com/sindresorhus/got/issues/997).
So upgrading [got](https://github.com/sindresorhus/got/) to version  >= v10.1.0 and passing timeout, retry options to checkHttp function can resolve this issue.

### PR DESCRIPTION
**Pull request by himanshumehta1114**
Now, is-reachable won't hang on passing URL which resolves DNS but does not gives any response.

fixes #47

### CODE REVIEW

**Comment by himanshumehta1114** [comment]
https://github.com/sindresorhus/got/blob/cf4fdad553f4bec82a9972dda40f5762f8c70c6d/package.json#L10

The latest version of got supports node >= 10. So can we update node version in `is-reachable` as well ? 
Got is dependency of `is-reachable` so it fails on node v8.

**Comment by sindresorhus** [comment]
> So can we update node version in is-reachable as well ?

👍🏻

**Comment by himanshumehta1114** [comment]
@silverwind @sindresorhus can we merge this?

File: index.js [hunk 1]
```diff
@@ -13,10 +13,16 @@ const URL = require('url-parse');
...
-const checkHttp = async url => {
+const checkHttp = async (url, timeout) => {
 	let response;
 	try {
-		response = await got(url, {rejectUnauthorized: false});
+		response = await got(url, {
+			https: {
+				rejectUnauthorized: false
+			},
+			retry: 0,
```
Comment (jkff) [inline index.js:23]: Is there a reason why we need to disable retries here? This is a pretty draconian way to check that a URL is reachable - many failures, such as ECONNRESET, can be intermittent.

I hit this while using another CLI tool that uses this library and refuses to do anything if one of the services it wants is "unreachable".

**Pull request merged**